### DISCUSSION TITLE
Is there an event called when a window move from/to another monitor?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#15401] [General]
At least in Windows, is there something in Avalonia to detect monitor change?

Also is there a property to get the current Window handle? like in WPF?

Thanks!

**Comment by thevortexcloud** [comment]
> is there something in Avalonia to detect monitor change?

I don't think so. But you could probably do it yourself somewhat easily. Avalonia provides all the information you need to determine which monitor the window is on (using [`Screens`](https://reference.avaloniaui.net/api/Avalonia.Controls/Screens/)) and there is an [event to detect when a window moves.
](https://reference.avaloniaui.net/api/Avalonia.Controls/WindowBase/05797BCB) From memory: You may need to do some maths though to determine exactly which monitor the window is on, since Avalonia treats the coordinates of each screen as offsets from each other so they are effectively one giant display as far as Avalonia is concerned.


> Also is there a property to get the current Window handle? like in WPF?

You can get the native handle by using `Handle` on the Window.

https://reference.avaloniaui.net/api/Avalonia.Platform/IWindowBaseImpl/E559DD02

**Reply by dbriard** [reply]
Thanks a lot for that super complete answer! I think there is everything I need :)

**Reply by thevortexcloud** [reply]
You are welcome. Do keep in mind resolutions/displays may change at pretty much anytime without warning (such as from the user changing display settings, running driver updates in the background or the cable being pulled), so you may need to recalculate in various other situations. If you need more complete monitor information (the Avalonia screen class is fairly minimal and is directly based on the Winforms class of the same name), you will likely need to query display information yourself or use another library. For example there is a pretty good SDL2 binding available for C#. 

https://github.com/flibitijibibo/SDL2-CS

Which you can in theory use to detect monitor change events.

https://wiki.libsdl.org/SDL2/SDL_DisplayEvent

The main problem with SDL2 though is it's a C/C++ API so it can confuse the hell out of people who have never touched it before. But it's _extremely_ portable. So it will work on every platform Avalonia works on.

**Reply by dbriard** [reply]
I think I used that lib may be 20 years ago, wow! For the moment I only care about monitor change to use the colour calibration profile associated with monitor, but you right, resolution and especially Dpi may change too at any time. And portability can be a big plus for MacOS so I keep that idea in mind, thanks!

**Reply by thevortexcloud** [reply]
> I think I used that lib may be 20 years ago, wow! 

Haha, it's still getting updated. SDL3 is meant to be out soonish with lots of major new features.

> thanks

You are welcome!

**Comment by maxkatz6** [comment]
Only PositionChanged compared to the Screen boundaries, I would assume.
